File: app/src/main/java/com/example/android/miniweather/View/SettingsFragment.java
package com.example.android.miniweather.View;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.preference.PreferenceManager;
import android.support.annotation.Nullable;

import com.example.android.miniweather.R;

/**
 * Created by matev on 2017. 11. 27..
 */

public class SettingsFragment extends PreferenceFragment
        implements Preference.OnPreferenceChangeListener {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.settings_main);

        Preference favouriteCity = findPreference(getString(R.string.settings_favourite_city_key));
        SharedPreferences sharedPreferences =
                PreferenceManager.getDefaultSharedPreferences(favouriteCity.getContext());
        favouriteCity.setOnPreferenceChangeListener(this);
        onPreferenceChange(favouriteCity, sharedPreferences.getString(favouriteCity.getKey(), ""));

    }

        @Override
    public boolean onPreferenceChange(Preference preference, Object object) {
        String value = object.toString();
        preference.setSummary(value);
        return true;
    }
}

File: app/src/main/java/com/example/android/miniweather/View/SettingsFragment.kt
package com.example.android.miniweather.View

import android.os.Bundle

import android.preference.PreferenceManager
import android.support.v7.preference.Preference
import android.support.v7.preference.PreferenceFragmentCompat
import com.example.android.miniweather.R

class SettingsFragment : PreferenceFragmentCompat(), Preference.OnPreferenceChangeListener {

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        addPreferencesFromResource(R.xml.settings_main)

        val favouriteCity = findPreference(getString(R.string.settings_favourite_city_key))
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(favouriteCity.context)
        favouriteCity.onPreferenceChangeListener = this
        onPreferenceChange(favouriteCity, sharedPreferences.getString(favouriteCity.key, ""))
    }

    override fun onPreferenceChange(preference: Preference?, newValue: Any?): Boolean {
        val value: String = newValue.toString()
        preference!!.summary = value
        return true
    }
}
